staging/src/k8s.io/code-generator/kube_codegen.sh

```diff
@@ -568,7 +568,7 @@ function kube::codegen::gen_client() {
         fi
     done < <(
         ( kube::codegen::internal::grep -l --null \
-            -e '+genclient' \
+            -e '//\s*+genclient' \
             -r "${in_dir}${one_input_api}" \
             --include '*.go' \
```

Comment:
This is OK, but still feels dirty.  Maybe we should take the output of the first grep and pipe it to `xargs grep -v '^// Code generated by .* DO NOT EDIT.$` (exept not literlly xargs because xargs is dumb WRT maybe-empty input on some OSes)?

Or is that overkill?

Reply:
Thanks for the quick review @thockin. I was also thinking something along those lines you suggested but decided to suggest the simplest thing, to avoid breaking anything and not to complicate the shell script wizardry more 😅

If xargs is not an option would  adding one more while-read be a possible solution?

``` bash
        ( kube::codegen::internal::grep -l --null \
            -e '//\s*+genclient' \
            -r "${in_dir}${one_input_api}" \
            --include '*.go' \
            || true \
        ) | while read -r f; do grep -L -e  '^// Code generated by .* DO NOT EDIT.$' $f; done \
        | while read -r -d $'\0' F; do dirname "${F}"; done
```

Reply:
We don't need both.  Simply matching on `\s*` seems sufficient.  In fact, maybe we should grep for `^\s*//\s*+genclient` to really anchor it.

In fact, we should probably do the same in all the other places we use `internal::grep`, even though they are not a problem (yet).  What do you think?

Reply:
> we should grep for `^\s*//\s*+genclient` to really anchor it.

Better 👍
 
> In fact, we should probably do the same in all the other places we use `internal::grep`, even though they are not a problem (yet). What do you think?

Makes sense to me. I'll make the change and update the PR